examples/lighting-app/lighting-common/BUILD.gn

```diff
@@ -13,14 +13,19 @@
 # limitations under the License.
 
 import("//build_overrides/chip.gni")
+import("${chip_root}/src/app/chip_additional_args.gni")
 import("${chip_root}/src/app/chip_data_model.gni")
 
 config("config") {
   include_dirs = [ "include" ]
 }
 
 chip_data_model("lighting-common") {
-  zap_file = "lighting-app.zap"
+  if (chip_disable_unique_id == true) {
+    zap_file = "lighting-app-data-model-no-unique-id.zap"
```

Comment:
Where is this .zap file

Also I think Boris is suggesting creating a new example examples/no-unique-id-app/

The reason why is if someone tries to build the all clusters app with the option `chip_disable_unique_id == true` that app will still have the UniqueId so the build flag is not doing what it is saying. By having a completely new example we have full control over what that app is supposed to do

Reply:
I've added new zap and matter files without `UniqueID` in them. I thought this was enough to remove support for `UniqueID`. What additional changes are required to completely remove support for `UniqueID`? Currently the chip-tool returns information that UniqueID is not supported. 

I wanted to remove the situation with duplicate lighting apps. I thought that only differences in the zap file were required and that I could just have two versions of that file in the repo, one with uniqueID and one without.

Reply:
>I've added new zap and matter files without UniqueID in them.

Very sorry I missed the files that was added somehow in my first review 

>I wanted to remove the situation with duplicate lighting apps.

That is Boris exact suggestion, why do you want to avoid that situation? It will be very clear what the intention of the example is.

The problem currently with `chip_disable_unique_id` (as it exists in this PR) is you have an application specific build flag that doesn't make sense for any other application. With the code you have today, I would expect that if I ran `./scripts/build/build_examples.py --target linux-x64-all-clusters-data-model-no-unique-id` that the application over there should also have no unique it, but only the zapfile of lighting will accept your `chip_disable_unique_id`

Reply:
ohhh, thanks for the clarification. In that case I will create another example application.

Reply:
@tehampson I created another example app as you wished for. 
About `chip_disable_unique_id` it will not works for all-clusters app at all. The additional argument to gn can only be passed if light-app is compiled. I tried adding `no-unique-id` for `all-clusters` and this is what I've got:
```
 ± ./scripts/build/build_examples.py  --pw-command-launcher=ccache  --target linux-x64-all-clusters-no-wifi-data-model-no-unique-id build
INFO    Building targets: linux-x64-all-clusters-no-wifi-data-model-no-unique-id
INFO    Preparing builder 'linux-x64-all-clusters-no-wifi-data-model-no-unique-id'
INFO    Generating /mnt/dane/projects/matter/connectedhomeip-third/out/linux-x64-all-clusters-no-wifi-data-model-no-unique-id
INFO    Generating linux-x64-all-clusters-no-wifi-data-model-no-unique-id
INFO    ERROR at the command-line "--args":1:76: Build argument has no effect.
INFO    pw_command_launcher="ccache" chip_enable_wifi=false chip_disable_unique_id=true
INFO                                                                               ^---
INFO    The variable "chip_disable_unique_id" was set as a build argument
INFO    but never appeared in a declare_args() block in any buildfile.
```

Reply:
>  The additional argument to gn can only be passed if light-app is compiled. I tried adding no-unique-id for all-clusters and this is what I've got:

Yes, but unfortunately the error looks like an error, not a feature :)

And duplicating the app does not necessarily mean that we have to duplicate source files. I think that on the build system level most file can be used directly from the lighting-app, only the zap needs to be explicitly specified for that new app.